recipes-graphics/wayland/weston_8.0.0.imx.bb

```diff
@@ -1,9 +1,6 @@
+require recipes-graphics/wayland/weston_8.0.0.bb
+
```

Comment:
We cannot link the files; you need to copy the patches and other files that are included on `SRC_URI`. Linking enforces a specific layout for the platform and we should never enforce that.

One pattern you could use is add modified or new patches/files under an `imx-specific` subdir and use this as part of `SRC_URI`.

Reply:
Is it common to move layers to custom locations? At least with a linked patch file, the use of a custom location would generate an immediate build error with a simple fix. With a copied patch file, there is no simple way to prevent our patch files from going stale.

Reply:
> 
> 
> Is it common to move layers to custom locations?

No, but recipe get upgraded via rename. When `require` is used - this would effectively break parsing stage once upstream gets upgraded (via `git mv ...`)

> At least with a linked patch file, the use of a custom location would generate an immediate build error with a simple fix. 

Immediate build failure - yes, but I'm not sure about an easy fix. 

> With a copied patch file, there is no simple way to prevent our patch files from going stale.

That I guess that is the whole intention of the fork - we get to keep it to ourselves, with all the goodies and baddies. Patch would never go stale on the fork, as it does not advance together with upstream. But even when it does - then it is the responsibility of the person upgrading the fork to upgrade patches.

Besides, who prevents from taking those patches submitted in oe-core layer and directly applying them to the fork?

Reply:
> > Is it common to move layers to custom locations?
> 
> No, but recipe get upgraded via rename. When `require` is used - this would effectively break parsing stage once upstream gets upgraded (via `git mv ...`)

Actually, the `require` is the one part that would just work because the require is a relative path.

> 
> > At least with a linked patch file, the use of a custom location would generate an immediate build error with a simple fix.
> 
> Immediate build failure - yes, but I'm not sure about an easy fix.

Okay, maybe not exactly easy. I was thinking about the user at least knowing about his custom layout, but finding the broken link might take some luck/insight on the user's part. But if the custom layout is uncommon, than to me it's still the lesser evil compared to not picking up some fix or feature from a modified upstream patch.

> 
> > With a copied patch file, there is no simple way to prevent our patch files from going stale.
> 
> That I guess that is the whole intention of the fork - we get to keep it to ourselves, with all the goodies and baddies. Patch would never go stale on the fork, as it does not advance together with upstream. But even when it does - then it is the responsibility of the person upgrading the fork to upgrade patches.
> 
> Besides, who prevents from taking those patches submitted in oe-core layer and directly applying them to the fork?

Sure, there's nothing preventing someone from monitoring the upstream changes and updating our patches. The problem is what do you do to make sure that it gets done in a timely fashion? With my proposal there is zero effort required.